pkg/api/v1/server.go

```diff
@@ -54,7 +55,13 @@ func (s *Server) fromDBModel(dbS *models.Server) error {
 	s.FacilityCode = dbS.FacilityCode.String
 	s.CreatedAt = dbS.CreatedAt.Time
 	s.UpdatedAt = dbS.UpdatedAt.Time
-	s.DeletedAt = dbS.DeletedAt.Time
+
+	nullTime := null.Time{}.Time
+	if dbS.DeletedAt.Time == nullTime {
+		s.DeletedAt = nil
+	} else {
+		s.DeletedAt = &dbS.DeletedAt.Time
+	}
```

Comment:
`DeletedAt` already starts as a nil so no need to set it to one. Also our null package provides us `IsZero()` which lets us check if it is set pretty easily. 

```suggestion
	if !dbS.DeletedAt.IsZero() {
		s.DeletedAt = &dbS.DeletedAt.Time
	}
```